apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/dot-template-create-edit.component.ts

```diff
@@ -76,26 +77,28 @@ export class DotTemplateCreateEditComponent implements OnInit, OnDestroy {
     }
 
     /**
-     * Save template to store
+     * Save and publish template to store
      *
+     * @param DotTemplate template
      * @memberof DotTemplateCreateEditComponent
      */
-    saveTemplate({ layout, body, themeId }: DotTemplate): void {
-        let value = {
+    saveAndPublishTemplate(template: DotTemplate): void {
+        this.store.saveAndPublishTemplate({
             ...this.form.value,
-            body
-        };
+            ...this.formatTemplateItem(template)
+        });
+    }
 
-        if (layout) {
-            value = {
-                ...this.form.value,
-                layout,
-                theme: themeId
-            };
-        }
+    /**
+     * Save template to store
+     *
+     * @param DotTemplate template
+     * @memberof DotTemplateCreateEditComponent
+     */
+    saveTemplate(template: DotTemplate): void {
```

Comment:
@rjvelazco have been working on some changes in all this, please check with him to avoid conflicts. Also check his PR.